### ISSUE TITLE
[Rappid] source.getPortSelector is not a function

### ISSUE THREAD

**Issue by tanakaworld** [clientIO/joint#569]
Hello!

I got an error `Uncaught TypeError: source.getPortSelector is not a function` in rappid v2.0.
I use to use it in JointJs v1.0(not rappid).
Is this method removed in rappid?

**Comment by kumilingus** [comment]
No, JointJS is a subset of Rappid.
Is `source` instance of `joint.shapes.devs.Model`? Are you sure you have used JointJS `v1.0` and not an older version? We have introduced new port API on `joint.dia.Element` in `v1.0` and `getPortSelector` was not needed anymore and therefore removed.

**Comment by tanakaworld** [comment]
The error was the cause using [v9.x interface](http://resources.jointjs.com/docs/jointjs/v0.9/joint.html#dia.Link). I solved it by using [v1.1 interface](http://resources.jointjs.com/docs/jointjs/v1.1/joint.html#dia.Link.intro).

I used JointJS v0.9.7 :(
"source" is the instance from `joint.dia.Graph.prototype.getCell` 
I have used port selector in order to make the instance of joint.dia.Link.

```javascript
const source = graph.getCell('source-id');
const target = graph.getCell('target-id');

•••

new joint.dia.Link({
        source: { id: source.id, selector:  source.getPortSelector('source-port-id')},
        target: { id: 'target.id, selector: target.getPortSelector('target-port-id')'}
});
```

I updated to following code.

```javascript
const source = graph.getCell('source-id');
const target = graph.getCell('target-id');

•••

new joint.dia.Link({
        source: { id: source.id, port:  'source-port-id'},
        target: { id: target.id, port: 'target-port-id'}
});
```


Thank you.
